app/controllers/security_controller.rb

```diff
@@ -11,6 +11,34 @@ def show
 
   def report; end
 
+  def paginated_activity
+    page_number = params[:page_number].to_i
```

Comment:
What happens if ?page_number isn't in the URL?
Are you calling `.to_i` on nil at that point?

Could do `params.fetch(:page_number, 1).to_i` might guarentee there's a page and you're at the start if not stated?

Reply:
or is it zero indexed 🤔 ?

Reply:
This param comes from a [route segment](https://github.com/alphagov/govuk-account-manager-prototype/pull/614/files#diff-959bc9abc46a55332bb64d5155a79323afa75a50ec1a2137ddd22d926f62c6c5R29), so the user will see a 404 error if they don't have a page number as part of the URL.

If the page is out of bounds, we show a message explaining this.  Users could only possibly see this if they start tampering with the URL.

The one exception is entering zero.  This isn't zero-indexed (it is one-indexed, not really sure what the right name for it is, but it starts on 1), but when you enter the page number as 0, kaminari quietly treats this as having entered 1 (i.e. they see the first page of results and the link to the next page will link to page 2).

If the user changes this segment to anything that doesn't contain a number, `to_i` will return zero, so the user will always see the first page.

Reply:
Ok ace! That sounds like it's not a problem users should run into.
Thanks so much for the detailed explaination - makes a sense now.